.github/pull_request_template.md

```diff
@@ -0,0 +1,19 @@
+### Description
+Describe why you're making this change, in plain English.
+
+### Testing & Reproduction steps
+* In the case of bugs, describe how to replicate
+* If any manual tests were done, document the steps and the conditions to replicate
+* Call out any important/ relevant unit tests, e2e tests or integration tests you have added or are adding
+
+### Links
+Include any links here that might be helpful for people reviewing your PR (Asana tasks, GH issues, API docs, external benchmarks, tools docs, etc). If there are none, feel free to delete this section.
+
+Please be mindful not to leak any customer or confidential information. Feel free to use [HashiLinks](https://go.hashi.co/) to obfuscate links as needed.
```

Comment:
It's worth considering that the audience for this template is broader than just the Consul team, and also includes external contributors (who won't know what HashiLinks is, for example).

```suggestion
Include any links here that might be helpful for people reviewing your PR (Tickets, GH issues, API docs, external benchmarks, tools docs, etc). If there are none, feel free to delete this section.

Please be mindful not to leak any customer or confidential information. HashiCorp employees may want to use our internal URL shortener to obfuscate links.
```